modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -596,14 +624,22 @@ private CompletableFuture<AsyncSqlCursor<InternalSqlRow>> queryScript(
     ) {
         String schemaName = properties.get(QueryProperty.DEFAULT_SCHEMA);
         ZoneId timeZoneId = properties.get(QueryProperty.TIME_ZONE_ID);
+        Long queryTimeout = properties.getOrDefault(QueryProperty.QUERY_TIMEOUT, 0L);
+
+        Instant deadline;
+        if (queryTimeout != 0) {
+            deadline = Instant.now().plusMillis(queryTimeout);
```

Comment:
what reason to use Instant instead of simple long and System.currentTimeMillis (in fact it should be FastTimestamps.coarseCurrentTimeMillis() )
The comment is common for the approach in the patch

Reply:
Personally, I would prefer to keep Instant. :thinking: 
From my point of view, the code is cleaner this way. But let's discuss this privately :sunglasses: